src/main/host/process.rs

```diff
@@ -1219,7 +1207,12 @@ impl Process {
 
 impl Drop for Process {
     fn drop(&mut self) {
-        unsafe { cshadow::process_free(self.cprocess.ptr()) }
+        // TODO: is there a better way to handle this?
+        // Adding `safely_drop` to Process itself is awkward; it'd need to take
+        // the RootedRc and determine whether the Process itself was being
+        // dropped or not.
+        Worker::with_active_host(|host| self.weak_rc.take().unwrap().safely_drop(host.root()))
+            .unwrap();
```

Comment:
I was worried about having the `RootedRc<RefCell<Process>>::safely_drop` drop the `RefCell<Process>`, which would call `RootedRcWeak<RefCell<Process>>::safely_drop` while we're still in the middle of running the previous `RootedRc::safely_drop` (and having two functions accessing the `RootedRcCommon` at the same time), but it looks to me like this works out properly. Maybe this would be worth having a unit test for in `rc.rs` (having a `RootedRc<T>` where `T` contains a `RootedRcWeak<T>` to itself)?